test/test_goodmans_er.py

```diff
@@ -1,6 +1,76 @@
 """Test Goodmans Ecological Regression."""
+import numpy as np
+import pytest
 from pyei.goodmans_er import *  # pylint:disable=wildcard-import,unused-wildcard-import
 
 
-def test_true():
-    assert True
+@pytest.fixture
+def group_and_vote_fractions():
+    """ Sample group and vote fractions, where every member of the demographic
+        group votes for the given candidate and every non-member of the
+        demographic group does not vote for the given candidate.
+    """
+    group_share = np.array([0, 0.2, 0.4, 0.6, 0.8])
+    vote_share = np.array([0, 0.2, 0.4, 0.6, 0.8])
+    return group_share, vote_share
+
+
+@pytest.fixture
+def group_and_vote_fractions_with_pop():
+    """ Sample group and vote fractions, where every member of the demographic
+        group votes for the given candidate and 10% of the demographic group's
+        complement supports the given candidate (i.e., slope = 1, intercept = 0.1),
+        with an exception of one precinct.
+        All precincts have population 1000, except one precinct that has population
+        1 and does not follow the above formula.
+    """
+    group_share = np.array([0, 0.2, 0.4, 0.6, 0.8, 0.9])
+    vote_share = np.array([0.1, 0.3, 0.5, 0.7, 0.9, 0.9])
+    populations = np.array([1000, 1000, 1000, 1000, 1000, 1])
+    return group_share, vote_share, populations
+
+
+def test_fit(group_and_vote_fractions):  # pylint:disable=redefined-outer-name
+    model = GoodmansER()
+    group_share, vote_share = group_and_vote_fractions
+    model.fit(group_share, vote_share)
+    np.testing.assert_almost_equal(model.intercept_, 0)
+    np.testing.assert_almost_equal(model.slope_, 1)
+
+
+def test_weighted_fit(group_and_vote_fractions_with_pop):  # pylint:disable=redefined-outer-name
+    model = GoodmansER(is_weighted_regression=True)
+    group_share, vote_share, pops = group_and_vote_fractions_with_pop
+    model.fit(group_share, vote_share, pops)
+    np.testing.assert_almost_equal(model.intercept_, 0.1, decimal=3)
+    np.testing.assert_almost_equal(model.slope_, 1, decimal=3)
+
+
+def test_summary():
+    model = GoodmansER()
+    model.demographic_group_name = "Trees"
+    model.candidate_name = "Lorax"
+    model.voting_prefs_est_ = 1.0
+    model.voting_prefs_complement_est_ = 0.0
+    assert (
```

Comment:
this block looks sort of terrible (the price of `black`!) I don't really care, but you could define the expected output as a separate variable:

```python
expected_summary = """Goodmans ER
Est. fraction of Trees...
"""
assert model.summary() == expected_summary
```

Reply:
Ooh, that's much cleaner! Thanks for the suggestion 👍